### PR TITLE
Deploy to k8s

### LINKED ISSUE #18: Kubernetes/tilt manifests and container images

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #18 

This PR migrates the JS build from pnpm to yarn, since [pnpm2nix](https://github.com/nix-community/pnpm2nix/) is pretty bitrotted, and neither [dream2nix](https://github.com/nix-community/dream2nix) nor [js2nix](https://github.com/canva-public/js2nix/) support pnpm.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -230,38 +231,38 @@ jobs:
...
-  #       run: make regenerate-charts
-  #     - name: Check if committed charts were up to date
-  #       run: git diff --exit-code
-  #     - name: Git Diff showed uncommitted changes
-  #       if: ${{ failure() }}
-  #       uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0 # v6
-  #       with:
-  #         script: |
-  #           core.setFailed('Committed charts were not up to date, please regenerate and re-commit!')
+  check_charts:
```
Comment (Techassi) [inline .github/workflows/build.yml:234]: It might be worth to cover this as a pre-commit hook.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -272,72 +273,72 @@ jobs:
...
-  #       # "-pr<prnumber>" so that the published artifacts can be linked to this PR.
-  #     - name: Update version if PR
-  #       if: ${{ github.event_name == 'pull_request' }}
-  #       run: python/cargo_version.py -m pr${{ github.event.pull_request.number }}
+        # This step checks if the current run was triggered by a push to a pr (or a pr being created).
+        # If this is the case it changes the version of this project in all Cargo.toml files to include the suffix
+        # "-pr<prnumber>" so that the published artifacts can be linked to this PR.
+      - name: Update version if PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: python/cargo_version.py -m pr${{ github.event.pull_request.number }}
```
Comment (Techassi) [inline .github/workflows/build.yml:318]: This will fail, right? The referenced file doesn't seem to be included in this PR.

File: deploy/helm/stackable-cockpit/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+make compile-chart
+
+helm install stackable-cockpit deploy/helm/stackable-cockpit
+```
+
+## Usage of the CRDs
+
+The usage of this operator and its CRDs is described in the [documentation](https://docs.stackable.tech/stackable-cockpit/index.html)
+
+The operator has example requests included in the [`/examples`](https://github.com/stackabletech/stackable-cockpit/tree/main/examples) directory.
```
Comment (Techassi) [inline deploy/helm/stackable-cockpit/README.md:23]: Both these statements are currently not correct.

File: web/build.rs [hunk 4]
```diff
@@ -7,7 +7,7 @@ fn main() {
     // Vite always seems to update the mtime of the project folder on each build, so track the files that we know matter individually
```
Comment (Techassi) [inline web/build.rs:7]: Update for yarn.

File: web/build.rs [hunk 5]
```diff
@@ -16,7 +16,7 @@ fn main() {
     ] {
         println!("cargo:rerun-if-changed={tracked_file}");
     }
-    let vite_status = Command::new("pnpm")
+    let vite_status = Command::new("yarn")
```
Comment (Techassi) [inline web/build.rs:19]:
```suggestion
    let yarn_status = Command::new("yarn")
```

Also needs update a little further down.

**Review by Techassi — COMMENTED** [review]
Partial review

File: .github/workflows/build.yml (hunk 1 above)
Comment (nightkr) [reply .github/workflows/build.yml:234]: This is pulled in from templating. In isolation I'd agree though.

File: .github/workflows/build.yml (hunk 2 above)
Comment (nightkr) [reply .github/workflows/build.yml:318]: Ah good catch.

File: deploy/helm/stackable-cockpit/README.md (hunk 3 above)
Comment (nightkr) [reply deploy/helm/stackable-cockpit/README.md]: Yeah.. this came from templating. Removed in 33bf6f94cbad18c632b66e9c4739f9b4906cdd6c

File: web/build.rs (hunk 4 above)
Comment (nightkr) [reply web/build.rs:7]: Yarn replaces pnpm, not vite.

File: web/build.rs (hunk 5 above)
Comment (nightkr) [reply web/build.rs:19]: Yarn replaces pnpm, not vite.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+DOCKER_REPO := docker.stackable.tech
+ORGANIZATION := stackable
+# this will be overwritten by an environmental variable if called from the github action
+HELM_REPO := https://repo.stackable.tech/repository/helm-dev
+HELM_CHART_ARTIFACT := target/helm/${OPERATOR_NAME}-${VERSION}.tgz
+
+SHELL=/usr/bin/env bash -euo pipefail
+
+render-readme:
+	scripts/render_readme.sh
```
Comment (Techassi) [inline Makefile:25]: I don't think we need this. Also this will fail because the script `render_readme.sh` is not part of this PR.

File: web/build.rs (hunk 4 above)
Comment (Techassi) [reply web/build.rs:7]: Whoops, my bad. You are correct.

File: Makefile (hunk 6 above)
Comment (nightkr) [reply Makefile:25]: Also pulled in from templating but not currently invoked anywhere, and I assume we'll want to start using it _eventually_

File: .github/workflows/build.yml (hunk 2 above)
Comment (nightkr) [reply .github/workflows/build.yml]: e5da0747ff7f4f5e34713955aca5d2ad224fc7b7

File: .github/workflows/build.yml (hunk 1 above)
Comment (nightkr) [reply .github/workflows/build.yml:242]: I'd say file an issue on templating about it

File: Makefile (hunk 6 above)
Comment (Techassi) [reply Makefile:25]: Okay, so we have two options:

- Remove above lines from the Makefile so that we cannot invoke the failing command
- Or we add the missing script to this PR to make it run-able

Just leaving it as is seems a little weird.

Comment (nightkr) [reply Makefile:25]: I'll comment it out

Comment (nightkr) [reply Makefile]: 010950b2441b682e1d6857a7c29ce8e50bdd9e3a

File: .github/workflows/build.yml (hunk 1 above)
Comment (Techassi) [reply .github/workflows/build.yml:242]: Okay I will open an issue.

**Review by Techassi — APPROVED** [review]
Approved, but as already discussed, the build/deploy toolchain seems very convoluted. This discussion is, however, better placed in an arch meeting.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Techassi) [reply .github/workflows/build.yml:242]: Opened an issue here: https://github.com/stackabletech/operator-templating/issues/263
